crypto/stack/stack.c

```diff
@@ -167,6 +173,15 @@ static int sk_reserve(OPENSSL_STACK *st, int n, int exact)
     num_alloc = st->num + n;
     if (num_alloc < min_nodes)
         num_alloc = min_nodes;
+    
+    if (st->data == NULL) { /* Time to do (postponed) allocation */
+        /* AFAIK, num_alloc and st->num were 0, so num_alloc = min_nodes here */
```

Comment:
No, `num_alloc = min_nodes` is not the only option. If you OPENSSL_sk_reserve with larger value, num_alloc will be whatever you pass (but not larger than max_nodes).

Reply:
I was trying to avoid multi-line comments ;)
I get your point and so I enhance some comments.